File: Assets/Scripts/Dodge.js
#pragma strict

var destination : Transform;
private var originalPosition : Vector3;
var moveSpeed : float = .5;
private var timer : float = 0;
 
function Start () {
	if (Options.diff != 0)
	{
		moveSpeed = moveSpeed / Options.diff;
	}
  originalPosition = transform.position;  // grabs original position
}
 
function Update () {
	timer += Time.deltaTime;
  transform.position = Vector3.Lerp(originalPosition, destination.position, Mathf.PingPong(timer * moveSpeed, 1.0));  //moves object back and forth
}


function OnTriggerEnter(other : Collider) {  //destroys player if it touches
  if (other.gameObject.CompareTag("Player")) {
    //add change color, loud sound, and explode then timer to load
    other.gameObject.SetActive (false);
    SceneLoader.ReloadScene();
  }
}

File: Assets/Scripts/Dodge.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Dodge : MonoBehaviour
{
    public Transform destination;
    private Vector3 originalPosition;
    public float moveSpeed;
    private float timer;
    public virtual void Start()
    {
        if (Optionz.diff != 0)
        {
            this.moveSpeed = (float) (this.moveSpeed / Optionz.diff);
        }
        this.originalPosition = this.transform.position; // grabs original position
    }

    public virtual void Update()
    {
        this.timer = this.timer + Time.deltaTime;
        this.transform.position = Vector3.Lerp(this.originalPosition, this.destination.position, Mathf.PingPong(this.timer * this.moveSpeed, 1f)); //moves object back and forth
    }

    public virtual void OnTriggerEnter(Collider other) //destroys player if it touches
    {
        if (other.gameObject.CompareTag("Player"))
        {
            //add change color, loud sound, and explode then timer to load
            other.gameObject.SetActive(false);
            
            if (string.IsNullOrEmpty(SceneLoader.currentScene) || SceneLoader.currentScene == "Active Main Menu")
            {
                SceneLoader.ReloadScene();
            }
            else
            {
                SceneLoader.GameOver();
            }
        }
    }

    public Dodge()
    {
        this.moveSpeed = 0.5f;
    }

}